staging/src/k8s.io/apiserver/pkg/endpoints/handlers/fieldmanager/equality.go

```diff
@@ -0,0 +1,62 @@
+/*
+Copyright 2021 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package fieldmanager
+
+import (
+	"context"
+	"reflect"
+
+	"k8s.io/apimachinery/pkg/api/equality"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/conversion"
+	"k8s.io/apimachinery/pkg/runtime"
+)
+
+var equalities = func() conversion.Equalities {
+	e := equality.Semantic
+	e.AddFunc(ignoreManagedFieldsTimestamp)
+	return e
+}()
+
+// EqualObjects returns true if the two objects are "equals".
+//
+// The two objects are equal if they have the exact same values, and
+// their managed fieldsets are the same, except for the timestamp.
+func EqualObjects(new, old runtime.Object) bool {
+	return equalities.DeepEqual(new, old)
+}
+
+// DropEqualTransformer is a transformer that cancels the change if the
+// objects are equal.
+func DropEqualTransformer(_ context.Context, newObj, oldObj runtime.Object) (runtime.Object, error) {
+	if EqualObjects(newObj, oldObj) {
+		return oldObj, nil
+	}
+	return newObj, nil
+}
+
+func ignoreManagedFieldsTimestamp(a, b metav1.ManagedFieldsEntry) bool {
+	aTime := a.Time
+	bTime := b.Time
+	a.Time = nil
+	b.Time = nil
+
+	equal := reflect.DeepEqual(a, b)
+	a.Time = aTime
+	b.Time = bTime
```

Comment:
shouldn't be necessary because you only modified a local copy, no?

Reply:
I'm not sure it's a copy in that context. These are the new and old objects, which aren't deep-copied.
Also in general, I'd rather have this function side-effect free.

Reply:
There's a shallow copy and the modification should affect only that shallow copy. If you're worried about someone in the future accidentally making a mutation outside of the shallow copy, write a test that would detect it? And that test would demonstrate that you don't need these lines.

Reply:
Yeah, this is absolutely a copy of the `ManagedFieldsEntry`, you're right 🤦